conda/recipes/libraft/recipe.yaml

```diff
@@ -78,6 +78,8 @@ cache:
           - libcusolver-dev
           - libcusparse-dev
       - librmm =${{ minor_version }}
+      - nccl ${{ nccl_version }}
+      - ucxx ${{ ucxx_version }}
```

Comment:
`libraft` recipe installs the `distributed` component https://github.com/rapidsai/raft/blob/902ed56969f588cf504bd9e85ee8ee7419812c93/conda/recipes/libraft/recipe.yaml#L94 which has a build and runtime dependency on targets from `NCCL` and `ucxx` as seen here https://github.com/rapidsai/raft/blob/902ed56969f588cf504bd9e85ee8ee7419812c93/cpp/CMakeLists.txt#L392-L406

I think these changes are needed and I believe the reason it did not error out until now is because `distributed` is an `INTERFACE` target so if consumer libraries brought in NCCL themselves it would be okay. 

Happy to revert the changes, however, if this is considered an improper way of working with header only packages.

Reply:
@divyegala I think these changes to the rattler-build recipe are reasonable, but can we break them out into a separate PR so we've got a cleaner history of when and why this was added?

Reply:
Agreed, this seems fine but doesn't make sense to bundle with the NCCL wheels work.

Reply:
Reverted the commit, opened new PR https://github.com/rapidsai/raft/pull/2636